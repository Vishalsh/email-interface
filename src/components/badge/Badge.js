import React from 'react';
import PropTypes from 'prop-types';

import classes from './Badge.module.scss';

const Badge = (props) => {
  const {
    label,
    className
  } = props;

  return (
    <span className={`${classes.badge} ${className}`}>{label}</span>
  );
};

Badge.defaultProps = {
  label: '',
  className: ''
};

Badge.propTypes = {
  label: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  className: PropTypes.string.isRequired
};

export default Badge;
