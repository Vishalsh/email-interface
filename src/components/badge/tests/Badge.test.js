import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Badge from '../Badge';

describe('Badge', () => {
  const props = {
    className: 'someClass',
    label: '10',
  };
  const component = shallow(<Badge {...props} />);

  it('should render the component', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
});