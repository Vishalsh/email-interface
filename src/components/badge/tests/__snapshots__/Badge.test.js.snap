// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render the component 1`] = `
<span
  className="badge someClass"
>
  10
</span>
`;
